core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -100,6 +113,11 @@ public void releaseAllListeners() {
   }
 
   @Override
+  public void setStoreListenerSource(Cache<K, V> source) {
+    storeListener.setSource(source);
+  }
+
```

Comment:
Even though this is unadvised, due to the absence of a better way to register ehcache instance to storeListener, a setSource method has been used here.